src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.HttpContentAsyncStream.cs

```diff
@@ -142,6 +149,15 @@ public override Task<int> ReadAsync(byte[] buffer, int offset, int count, Cancel
                 }
             }
 
+            public override void Write(byte[] buffer, int offset, int count)
+            {
+                // WriteAsync should be how writes are performed on this stream as
+                // part of HttpContent.CopyToAsync.  However, we implement Write just
+                // in case someone does need to do a synchronous write or has existing
+                // code that does so.
+                WriteAsync(buffer, offset, count, CancellationToken.None).GetAwaiter().GetResult();
```

Comment:
> There's a nice article about it here. :stuck_out_tongue:

Thanks.  Not sure the author really knows what he's talking about, though.

> Could this deadlock if there's a single threaded SynchronizationContext?

Nothing about the implementation should cause that kind of deadlock.  The implementation never schedules anything back to the original SynchronizationContext.

It could, however, block for an arbitrarily long period (it's pushing data which won't be consumed and allow the Write to wake up until libcurl asks for the data), which is why I didn't implement this initially, and why WCF shouldn't be using it :smile: